broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/StreamProcessorTransitionStep.java

```diff
@@ -129,14 +129,9 @@ private static StreamProcessor createStreamProcessor(
     final StreamProcessorMode streamProcessorMode =
         targetRole == Role.LEADER ? StreamProcessorMode.PROCESSING : StreamProcessorMode.REPLAY;
 
-    final var isBackupFeatureEnabled =
-        context.getBrokerCfg().getExperimental().getFeatures().isEnableBackup();
-
     final Engine engine = new Engine(context.getTypedRecordProcessorFactory());
     final List<RecordProcessor> recordProcessors =
-        isBackupFeatureEnabled
-            ? List.of(engine, context.getCheckpointProcessor())
```

Comment:
:question: Does it make sense to install the processor, even if the store is not properly configured? :thinking:

Reply:
Yes. We have to ensure that `checkpointId` is updated even if no backup is taken. Due to misconfiguration, one broker may have configured backup store. It will then update checkpointId and take backup. The other broker, if it does not increment its checkpointId, can lead to inconsistent backups later. This incosistency can also happen after a restart - after a configuration change to backup store.